controllers/paddlejob_controller.go

```diff
@@ -111,6 +116,26 @@ func (r *PaddleJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	// scheduling with volcano
+	if r.Scheduling == schedulerNameVolcano && withVolcano(&pdj) {
+		pg := &volcano.PodGroup{}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(&pdj), pg); err != nil {
+			if apierrors.IsNotFound(err) {
+				pg = constructPodGroup(&pdj)
+				if err = ctrl.SetControllerReference(&pdj, pg, r.Scheme); err != nil {
+					log.Error(err, "make reference failed")
+					return ctrl.Result{Requeue: true}, nil
+				}
+				if err = r.createResource(ctx, &pdj, pg); err != nil {
+					log.Error(err, "create podgroup failed")
+				}
+				return ctrl.Result{RequeueAfter: time.Second}, nil
+			} else {
+				return ctrl.Result{Requeue: true}, nil
+			}
+		}
+	}
+
```

Comment:
for delaying creating pods, l think we should also check pg status util it is running or inqueue

Reply:
done